x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -528,24 +539,59 @@ private void findActiveApiKeys(final BoolQueryBuilder boolQuery, ActionListener<
             .request();
         securityIndex.checkIndexVersionThenExecute(listener::onFailure,
             () -> ScrollHelper.fetchAllByEntity(client, request, listener,
-                (SearchHit hit) -> hit.getId()));
+                        (SearchHit hit) -> {
+                            Map<String, Object> source = hit.getSourceAsMap();
+                            String name = (String) source.get("name");
+                            String id = hit.getId();
+                            Long creation = (Long) source.get("creation_time");
+                            Long expiration = (Long) source.get("expiration_time");
```

Comment:
shouldn't we check to see if source contains an `expiration_time` field?

Reply:
Or even be a little overcautious and check that all fields are there before we access them? Just a thought